src/connection.js

```diff
@@ -488,7 +488,9 @@ class Connection extends EventEmitter {
     this.createTokenStreamParser();
     this.inTransaction = false;
     this.transactionDescriptors = [new Buffer([0, 0, 0, 0, 0, 0, 0, 0])];
-    this.transitionTo(this.STATE.CONNECTING);
+    if (openConnection) {
+      this.open();
```

Comment:
Would you mind moving this to be the very last thing happening inside `constructor`?